packages/wallets/nufi/src/adapter.ts

```diff
@@ -0,0 +1,289 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    scopePollingDetectionStrategy,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface NufiEvents {
+    connect(): void;
+    disconnect(): void;
+}
+
+/** The type of the connector object injected by the NuFi extension.
+ *
+ * Note that the interface uses some objects from `@solana/web3.js`. It is
+ * guaranteed that NuFi uses at least version `1.42.0` of this library.
+ */
```

Comment:
In the interest of brevity, these comments can probably all be removed since they document a purely internal interface.

Reply:
I very much disagree. This is the interface between the wallet and the adapter, and it's anything but internal. I feel like a precise definition of this interface is important. And we don't have this interface documentation published anywhere else at the moment.

Nonetheless, I removed them for now.